src/gfn/containers/base.py

```diff
@@ -54,3 +54,11 @@ def load(self, path: str) -> None:
                 self.__dict__[key] = torch.load(os.path.join(path, key + ".pt"))
             else:
                 raise ValueError(f"Unexpected {key} of type {type(val)}")
+
+
+def has_log_probs(obj: Container):
+    """Returns True if the submitted container has the log_probs attribute populated."""
+    if not hasattr(obj, "log_probs"):
+        return False
+
+    return obj.log_probs is not None and obj.log_probs.nelement() > 0
```

Comment:
I'm curious to understand why chose it to be outside the class. Can't it be an instance method?

Reply:
I actually just moved this from another location in the code to avoid circular imports. You're right, it's even cleaner as a instance method of the base class.